src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,44 +1,127 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
     private static final int DEFAULT_CAPACITY = 16;
     private static final float LOAD_FACTOR = 0.75f;
-    private Node<K, V>[] buckets = new Node[DEFAULT_CAPACITY];
+    private Node<K, V>[] buckets;
+    private int capacity;
     private int size;
     private int threshold;
 
+    public MyHashMap() {
+        this.buckets = new Node[DEFAULT_CAPACITY];
+        this.capacity = DEFAULT_CAPACITY;
+        this.threshold = (int) (capacity * LOAD_FACTOR);
+    }
+
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (size >= threshold) {
+            grow();
+        }
+        int index = getIndexByKey(key);
+        if (buckets[index] == null) {
+            fillEmptyBucket(newNode, index);
+        } else {
+            addNextToBucket(newNode, index);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndexByKey(key);
+        V value;
+        Node<K, V> node = buckets[index];
+        if (node == null) {
+            return null;
+        }
+        value = node.value;
+        while (node.next != null) {
+            node = node.next;
+            if (Objects.equals(node.key, key)) {
+                value = node.value;
+            }
+        }
+        return value;
     }
 
     @Override
     public int getSize() {
         return size;
     }
 
-    private int hash() {
-        return 0;
+    private int getIndexByKey(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % capacity);
     }
 
-    private void grow(){}
+    private void fillEmptyBucket(Node<K, V> node, int index) {
+        buckets[index] = node;
+        size++;
+    }
+
+    private void addNextToBucket(Node<K, V> node, int index) {
+        Node<K, V> prev = buckets[index];
+        if (isEquals(node, prev)) {
+            node.next = prev.next;
+            buckets[index] = node;
+        } else {
+            while (prev.next != null) {
+                if (isEquals(node, prev.next)) {
+                    node.next = prev.next.next;
+                    prev.next = node;
+                    return;
+                }
+                prev = prev.next;
+            }
+            prev.next = node;
+            size++;
+        }
+    }
+
+    private boolean isEquals(Node<K, V> firstNode, Node<K, V> secondNode) {
+        return firstNode.hash == secondNode.hash && Objects.equals(firstNode.key, secondNode.key);
```

Comment:
if keys is equals then hashCode equals
key1 == key2 || key1 != 0 && key1.equals(key2)